pkg/asset/agent/manifests/infraenv.go

```diff
@@ -70,6 +70,10 @@ func (i *InfraEnv) Generate(_ context.Context, dependencies asset.Parents) error
 				i.Config.Spec.AdditionalNTPSources = agentConfig.Config.AdditionalNTPSources
 			}
 
+			if agentConfig.Config == nil && installConfig.Config.BareMetal.AdditionalNTPServers != nil {
```

Comment:
Maybe checking i.Config.Spec.AdditionalNTPSources instead would be better as thats what we're concerned about setting i.e.
> if i.Config.Spec.AdditionalNTPSources == nil && installConfig.Config.BareMetal.AdditionalNTPServers != nil {